version: "3.9"
services:

  postgres:
    image: postgres
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
      PGDATA: ${PGDATA}
    volumes:
      - ./pg-data:/var/lib/postgresql/data/pgdata
      - .:/postgres/docker-entrypoint-initdb.d
    networks:
      - db-network
    restart: on-failure
    ports:
      - "5432:5432"
  backend:
    build: ./backend
    ports:
      - "3000:3000"
    networks:
      - db-network
      - app-network
    restart: on-failure
    environment:
      DATABASE_DRIVER: ${DATABASE_DRIVER}
      DATABASE_URL: ${DATABASE_URL}
      DATABASE_USERNAME: ${DATABASE_USERNAME}
      DATABASE_PASSWORD: ${DATABASE_PASSWORD}
      DATABASE_NAME: ${DATABASE_NAME}
      DATABASE_PORT: ${DATABASE_PORT}
      DEBUG: ${DEBUG}
    depends_on:
      - postgres
  frontend:
    build: ./frontend
    ports:
      - "5173:5173"
    volumes:
      - ./frontend:/usr/src/app/dist
    networks:
      - app-network
    restart: on-failure

  nginx:
    build: ./nginx
    networks:
      - internet
      - app-network
    restart: on-failure
    depends_on:
      - backend
      - frontend
    ports:
      - "80:80"

networks:
  db-network:
    driver: bridge
    internal: true
  app-network:
    driver: bridge
    internal: true
  internet: 
    driver: bridge
    
volumes:
  frontend:
    external: true

    
